src/ansys/mapdl/core/cli/hpc.py

```diff
@@ -0,0 +1,271 @@
+)
+@click.option(
+    "--config_file",
+    default=None,
+    type=str,
+    help="""To load job configuration from a file.""",
+)
+@click.option(
+    "--num_cores",
+    default=None,
+    type=str,
+    help="""Set the amount of CPU cores reserved for the job. By default it is 1 CPU.""",
+)
+@click.option(
+    "--memory",
+    default=None,
+    type=str,
+    help="""Set the amount of memory RAM in MB reserved for the job. By default it is 100 MB.""",
+)
+@click.option(
+    "--disk_space",
+    default=None,
+    type=str,
+    help="""Set the amount of hard drive space in MB reserved for the job. By default it is 100 MB.""",
+)
+@click.option(
+    "--exclusive",
+    default=None,
+    type=str,
+    is_flag=False,
+    flag_value=True,
+    help="""Set the job to run in a machine exclusively, without sharing it with other jobs. By default it is False""",
+)
+@click.option(
+    "--max_execution_time",
+    default=None,
+    type=str,
+    help="""Set the maximum execution time for the job. By default it is zero (unlimited).""",
+)
+@click.option(
+    "--wait",
+    default=None,
+    type=str,
+    is_flag=False,
+    flag_value=True,
+    help="""Set the terminal to wait for job completion before return the control to the user. """,
+)
+@click.option(
+    "--save_config_file",
+    default=False,
+    type=bool,
+    is_flag=False,
+    flag_value=True,
+    help="""Writes the configuration to the config file, after successfully
+submit the job. It overwrites the configuration file.
+The configuration file path is given using ``config_file`` argument.""",
+)
+@click.option(
+    "--debug",
+    default=False,
+    type=bool,
+    is_flag=False,
+    flag_value=True,
+    help="""Set PyMAPDL to prints the debug logging to the console output.""",
+)
+def submit(
+    main_file: str,
+    name: str = None,
+    url: str = None,
+    user: str = None,
+    password: str = None,
+    python: float = 3.9,
+    output_files: Optional[Union[str, list]] = None,
+    shell_file: str = None,
+    requirements_file: str = None,
+    extra_files: Optional[Union[str, list]] = None,
+    config_file: str = None,
+    num_cores: int = None,
+    memory: int = None,
+    disk_space: int = None,
+    exclusive: bool = None,
+    max_execution_time: int = None,
+    wait: bool = False,
+    save_config_file: bool = False,
+    debug: bool = False,
+):
+    """Example code:
+    pymapdl submit my_file.sh my_file_01.py my_file_02  --name="my job" --url="https://10.231.106.91:3000/hps" --user=repuser --password=repuser --python=3.9
+    """
+    import json
+
+    from ansys.mapdl.core.hpc.pyhps import (
+        create_pymapdl_pyhps_job,
+        get_value_from_json_or_default,
+        wait_for_completion,
+    )
+
+    if debug:
+        logging.basicConfig(
+            format="[%(asctime)s | %(levelname)s] %(message)s", level=logging.DEBUG
+        )
+
+    if config_file is None:
+        config_file = os.path.join(os.getcwd(), "hps_config.json")
```

Comment:
@germa89 should the config be in `appdirs` instead?

Reply:
MMhhh... if we use a file in the root directory, we could point to different files.... and hence multiple configurations.

Reply:
But maybe you are right... maybe it is better to use ``appdirs``.... buttt... it might be a bit hidden for the user...

Reply:
And we have many options to be keep in the file:
```json
{"url": "https://10.231.106.91:3000/hps", "user": "repuser", "password": "repuser", "python": "3.9", "name": "my job", "num_cores": 1, "memory": 100, "disk_space": 100, "exclusive": false, "max_execution_time": 1000}
```
keeping so many important options hidden in a ``appdir`` path might bring some complications.

Reply:
@germa89 use keyring for user/password. I don't think its secure enough to put that in plaintext like this. For everything else I would agree

Reply:
I agree.